src/components/vr_cooperation/include/vr_cooperation/event_engine/event.h

```diff
@@ -37,6 +37,12 @@
 
 namespace event_engine {
 
+enum MessageType {
+  kRequest,
+  kResponse,
+  kNotification
+};
```

Comment:
I think it is enough. Event is base class. Derived class should implement event_message_type according with this enumeration. Client code using  event_engine should handle message from HMI (or mobile) and create valid Event objects or don't create at all.